content/en/blog/2022/cni-mode/index.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "Merbridge CNI Mode"
+linkTitle: "Merbridge CNI Mode"
+date: 2022-05-18
+weight: 1
+description: This blog explains how CNI works in Merbridge.
+---
+
+<!--Merbridge CNI 模式的出现，旨在能够更好地适配服务网格的功能。之前没有 CNI 模式时，Merbridge 能够做得事情比较有限。其中最大的问题是不能适配注入 Istio 的 Sidecar Annotation，这就导致 Merbridge 无法排除某些端口或 IP 段的流量等。同时，由于之前 Merbridge 只处理 Pod 内部的连接请求，这就导致，如果是外部发送到 Pod 的流量，Merbridge 将无法处理。-->
+The CNI mode is designed to make Merbridge work better with service mesh features. Before having this mode, Merbridge cannot exclude traffic from certain ports or IP ranges, because Istio Annotations configured for a pod cannot take effect in Merbridge. Besides, it can only process connection requests in pods. Traffic from external to pods cannot be processed.
+
+<!--为此，我们精心设计了 Merbridge CNI，旨在解决这些问题。-->
+Therefore, we developed the Merbridge CNI mode to address these issues.
+
+<!--## 为什么需要 CNI 模式？-->
+## Why CNI mode is needed
+<!--其一，之前的 Merbridge 只有一个很小的控制面，其监听 Pod 资源，将当前节点的 IP 信息写入 `local_pod_ips` 的 map，以供 connect 使用。但是，connect 程序由于工作在主机内核层，其无法知道当前正在处理的是哪个 Pod 的流量，就没法处理如 `excludeOutboundPorts` 等配置。为了能够适配注入 `excludeOutboundPorts` 的 Sidecar Annotation，我们需要让 eBPF 程序能够得知当前正在处理哪个 Pod 的请求。-->
+First, Merbridge has a small control plane, which listens to Pods and writes the current node IP into the map of `local_pod_ips` for `connect` to use. However, `connect` works in the host kernel. It needs to know a pod's identity before responding to configurations such as `excludeOutboundPorts`. In order to make Sidecar Annotations injected with `excludeOutboundPorts` effective, we need to let eBPF know the request is from which Pod.
```

Comment:
First, Merbridge has a small control plane, which listens to Pods and writes the current node IP into the map of `local_pod_ips` for `connect` to use. However, `connect` works in the host kernel. It needs to know a pod's identity before responding to configurations such as `excludeOutboundPorts`. In order to make Sidecar Annotations injected with `excludeOutboundPorts` effective, we need to let eBPF know which Pod the request is from.